dbcon/execplan/arithmeticoperator.h

```diff
@@ -238,6 +239,51 @@ inline void ArithmeticOperator::evaluate(rowgroup::Row& row, bool& isNull, Parse
       break;
 
     case execplan::CalpontSystemCatalog::UBIGINT:
+      {
+        // XXX: this is bandaid solution for specific customer case (MCOL-5568).
+        // Despite that I tried to implement a proper solution: to have operations
+        // performed using int128_t amd then check the result.
+        int128_t x, y;
+        bool signedLeft = lop->data()->resultType().isSignedInteger();
+        bool signedRight = rop->data()->resultType().isSignedInteger();
+        if (signedLeft)
+        {
+          x = (int128_t)lop->getIntVal(row, isNull);
```

Comment:
Did you check if this assignment generates MOVDQA, MOVDQU or MOV?
If it is MOVDQA and getIntValue dst is aligned by 8 bytes there will be SEGV here.